mcs/class/System/System.Diagnostics/Process.cs

```diff
@@ -98,7 +100,14 @@ private struct ProcInfo
 		[MonitoringDescription ("The main module of the process.")]
 		public ProcessModule MainModule {
 			get {
-				return(this.Modules[0]);
+				/* Optimize Process.GetCurrentProcess ().MainModule */
+				if (processId == NativeMethods.GetCurrentProcessId ()) {
+					if (current_main_module == null)
+						current_main_module = this.Modules [0];
+					return current_main_module;
+				} else {
+					return this.Modules [0];
```

Comment:
why aren't we caching `this.Modules [0]` in all cases? is there a way it could change?

Reply:
Because we are caching it in a static variable, so we can only store 1 value.

Reply:
ahh, I see 😄